src/main.js

```diff
@@ -284,21 +285,23 @@ function handleClose(streams, children, childrenInfo) {
         if (aliveChildren.length === 0) {
             exit(exitCodes);
         }
-    });
 
-    if (config.killOthers) {
```

Comment:
The problem was that this was outside of the `closeStream.subscribe` and there was no check as to whether the exit code was non-zero. This basically meant that `--kill-others` would kill all other commands after the first exits.

I've moved this to inside of `closeStream.subscribe` with a check against the `exitCode` (also had to add a check for whether `othersKilled` was set because async (don't want to sigterm other commands twice).